src/routes/choice-exercises/components/SingleChoiceExercise.tsx

```diff
@@ -81,16 +81,16 @@ const ChoiceExerciseScreen = ({
   }, [results, currentWord])
 
   const onExerciseFinished = async (results: DocumentResult[]): Promise<void> => {
-    if (disciplineId) {
-      await saveExerciseProgress(disciplineId, exerciseKey, results)
-    }
+    const progress = await getExerciseProgress()
+    await saveExerciseProgress(disciplineId, exerciseKey, results)
     navigation.navigate('ExerciseFinished', {
       documents,
       disciplineId,
       disciplineTitle,
       exercise: exerciseKey,
       results,
-      closeExerciseAction: route.params.closeExerciseAction
+      closeExerciseAction: route.params.closeExerciseAction,
+      unlockedNextExercise: progress[disciplineId] ? progress[disciplineId]![exerciseKey] === undefined : true
```

Comment:
If I understand this correctly this should be the same as
```suggestion
      unlockedNextExercise: progress[disciplineId]?.[exerciseKey] === undefined
```